embassy/src/blocking_mutex/mod.rs

```diff
@@ -1,14 +1,23 @@
-//! Blocking mutex (not async)
-
+//! Blocking mutex.
+//!
+//! This module provides a blocking mutex that can be used to synchronize data.
 pub mod raw;
 
 use core::cell::UnsafeCell;
 
 use self::raw::RawMutex;
 
-/// Any object implementing this trait guarantees exclusive access to the data contained
-/// within the mutex for the duration of the lock.
-/// Adapted from <https://github.com/rust-embedded/mutex-trait>.
+/// Blocking mutex (not async)
+///
+/// Provides a blocking mutual exclusion primitive backed by an implementation of [`raw::RawMutex`].
```

Comment:
One thing that Dario mentioned, which helped me a lot, was that the blocking mutex is designed for bringing your own interior mutability. They are also low level on this basis. In particular, locks are intended to be short lived and must not cross async boundaries (and their design enforces this too). OTOH, the async mutex can have a lock spanning awaits, but also comes with a little more overhead.

Reply:
Thanks, I've emphasized this point